src/lib.rs

```diff
@@ -155,16 +155,24 @@ impl CodecPipelineImpl {
     }
 
     fn py_untyped_array_to_array_object<'a>(
-        value: &Bound<'a, PyUntypedArray>,
+        value: &'a Bound<'_, PyUntypedArray>,
     ) -> &'a PyArrayObject {
         let array_object_ptr: *mut PyArrayObject = value.as_array_ptr();
-        unsafe {
-            // SAFETY: array_object_ptr cannot be null
-            &*array_object_ptr
-        }
+        let array_object: &'a PyArrayObject = unsafe {
+            // SAFETY: array_object_ptr cannot be null and the array object pointed to by array_object_ptr is valid for 'a
+            array_object_ptr
+                .as_ref()
+                .expect("pointer is convertible to a reference")
+        };
+        array_object
     }
 
-    fn nparray_to_slice<'a>(value: &'a Bound<'_, PyUntypedArray>) -> &'a [u8] {
+    fn nparray_to_slice<'a>(value: &'a Bound<'_, PyUntypedArray>) -> Result<&'a [u8], PyErr> {
+        if !value.is_c_contiguous() {
+            return Err(PyErr::new::<PyValueError, _>(
+                "input array must be a C contiguous array".to_string(),
+            ));
+        }
         let array_object: &PyArrayObject = Self::py_untyped_array_to_array_object(value);
         let array_data = array_object.data.cast::<u8>();
         let array_len = value.len() * Self::pyarray_itemsize(value);
```

Comment:
is the `* Self::pyarray_itemsize(value)` correct or should that be `* 1`? (same below)